velox/dwio/parquet/tests/ParquetTpchTest.cpp

```diff
@@ -0,0 +1,221 @@
+    parquet::registerParquetReaderFactory();
+  }
+
+  void TearDown() override {
+    parquet::unregisterParquetReaderFactory();
+    HiveConnectorTestBase::TearDown();
+  }
+
+  std::vector<std::string> getLineItemFilePaths() {
+    auto fileNames = {
+        "lineitem1.par", "lineitem2.par", "lineitem3.par", "lineitem4.par"};
+    std::vector<std::string> filePaths;
+    for (const auto& fileName : fileNames) {
+      filePaths.push_back(facebook::velox::test::getDataFilePath(
+          "velox/dwio/parquet/tests", fmt::format("tpch_tiny/{}", fileName)));
+    }
+    return filePaths;
+  }
+
+  std::shared_ptr<connector::hive::HiveConnectorSplit> makeSplit(
+      const std::string& filePath) {
+    auto split = makeHiveConnectorSplit(filePath);
+    split->fileFormat = dwio::common::FileFormat::PARQUET;
+    return split;
+  }
+
+  void addSplitsToTask(
+      exec::Task* task,
+      bool& noMoreSplits,
+      const std::vector<std::string>& filePaths,
+      int sourcePlanNodeId) {
+    if (!noMoreSplits) {
+      for (const auto& filePath : filePaths) {
+        task->addSplit(
+            std::to_string(sourcePlanNodeId), exec::Split(makeSplit(filePath)));
+      }
+      task->noMoreSplits(std::to_string(sourcePlanNodeId));
+      noMoreSplits = true;
+    }
+  }
+};
+
+TEST_F(ParquetTpchTest, tpchQ1) {
+  const auto& filePaths = getLineItemFilePaths();
+
+  auto rowType =
+      ROW({"returnflag",
+           "linestatus",
+           "quantity",
+           "extendedprice",
+           "discount",
+           "tax",
+           "shipdate"},
+          {VARCHAR(),
+           VARCHAR(),
+           DOUBLE(),
+           DOUBLE(),
+           DOUBLE(),
+           DOUBLE(),
+           VARCHAR()});
+
+  // date '1998-12-01' - interval '90' day <= shipdate <= '1998-12-01'
+  auto filters =
+      common::test::SubfieldFiltersBuilder()
+          .add("shipdate", common::test::between("1998-09-02", "1998-12-01"))
+          .build();
+
+  const int sourcePlanNodeId = 10;
+  static const core::SortOrder kAscNullsLast(true, false);
+
+  const auto stage1 = PlanBuilder(sourcePlanNodeId)
+                          .tableScan(
+                              rowType,
+                              makeTableHandle(std::move(filters)),
+                              allRegularColumns(rowType))
+                          .project(
+                              {"returnflag",
+                               "linestatus",
+                               "quantity",
+                               "extendedprice",
+                               "extendedprice * (1.0 - discount)",
+                               "extendedprice * (1.0 - discount) * (1.0 + tax)",
+                               "discount"})
+                          .partialAggregation(
+                              {0, 1},
+                              {"sum(quantity)",
+                               "sum(extendedprice)",
+                               "sum(p4)",
```

Comment:
nit: you as use "AS" to specify an alias, e.g. "extendedprice * (1 - discount) as sum_disc_price", then "sum(sum_disc_price)"